src/controller.ts

```diff
@@ -1,23 +1,73 @@
 import { Application } from "pixi.js";
 import { Minigame, MinigameDelegate } from "./minigames/minigame";
 import { KeyboardMinigame } from "./minigames/keyboardMinigame";
+import { Interlude, InterludeDelegate } from "./interlude/interlude";
 
+const ALL_MINIGAMES = [Minigame, KeyboardMinigame];
 /** Controls the flow of the game. */
-export class Controller implements MinigameDelegate {
-  private currentMinigame?: Minigame;
+export class Controller implements MinigameDelegate, InterludeDelegate {
+  private completedMinigamePhases = 0;
+
+  private minigameQueue: Minigame[] = [];
+  private minigameWinCount = 0;
+
+  private currentScreen?: Minigame | Interlude;
 
   constructor(private readonly app: Application) { }
 
   start() {
-    this.currentMinigame = new KeyboardMinigame(this.app, this);
-    this.currentMinigame.attach();
+    this.populateMinigameQueue(10);
+    this.startNextMinigame();
   }
 
   onMinigameEnd(passed: boolean) {
```

Comment:
Possible refactor?

```
let stage: 'in-minigame' | 'in-interlude' | 'game-over' | 'won';
// Update state.
if (passed) {
  this.minigameWinCount++;
}
if (this.minigameWinCount >= 5) {
    this.completedMinigamePhases++;
}
// Figure out stage.
if (this.completedMinigamePhases >= 3) {
    stage = 'won';
} else if (this.minigameWinCount >= 5) {
    this.minigameWinCount = 0;
    stage = 'in-interlude';
} else if (this.minigameQueue.length === 0) {
   // We ran out of minigames.
   stage = 'game-over';
} else {
   // keep trying.
   stage = 'in-minigame';
}

... Handle setting up next stage
```

Just a little bit more clear since we have the stage variable document what decisions the code is making. Optional tho cause this is a game jam

Reply:
I originally was going to put something like this but decided against it because it seemed a bit excessive for what probably isn't going to grow in complexity. We can make the refactor later if it seems useful